src/main/java/com/st/utopia/agent/controller/BookingController.java

```diff
@@ -0,0 +1,203 @@
+package com.st.utopia.agent.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+
+import com.st.utopia.agent.model.PaymentAmount;
+import com.st.utopia.agent.model.Ticket;
+import com.st.utopia.agent.model.User;
+
+/**
+ * Controller for agents using this user-facing interface
+ *
+ * @author Jonathan Lovelace
+ * @author Salem O
+ */
+@RestController
+@RequestMapping("/booking")
+public class BookingController {
```

Comment:
Personally, I wouldn't split things up by which backend they call, especially because in my Node implementation at least one route calls different backends depending on circumstances.